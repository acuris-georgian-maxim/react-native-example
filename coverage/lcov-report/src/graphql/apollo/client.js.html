<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/graphql/apollo/client.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/graphql/apollo</a> client.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/12</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
&nbsp;
import { ApolloClient } from 'apollo-client';
import { withClientState } from 'apollo-link-state';
import { ApolloLink } from 'apollo-link';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { HttpLink } from 'apollo-link-http';
&nbsp;
import { config } from '../config';
&nbsp;
const cache = new InMemoryCache({
    dataIdFromObject: object =&gt; object.key || null
});
&nbsp;
const defaults = {
    list: {
        listIsVisible: true,
        __typename: 'list'
    },
    menu: {
        menuIsCollapsed: false,
        __typename: 'menu'
    }
};
&nbsp;
const stateLink = withClientState({
    cache,
    defaults,
    resolvers: {
        Mutation: {
            /* eslint-disable */
            updateListVisibility: <span class="fstat-no" title="function not covered" >(_</span>, { listIsVisible }, { cache }) =&gt; {
                /* eslint-enabled */
                const data = <span class="cstat-no" title="statement not covered" >{</span>
                    list: {
                        listIsVisible,
                        __typename: 'list'
                    }
                };
&nbsp;
<span class="cstat-no" title="statement not covered" >                cache.writeData({ data });</span>
<span class="cstat-no" title="statement not covered" >                return null;</span>
            },
            updateMenuVisiblity: <span class="fstat-no" title="function not covered" >(_</span>, { menuIsCollapsed }, { cache }) =&gt; {
                /* eslint-enabled */
                const data = <span class="cstat-no" title="statement not covered" >{</span>
                    menu: {
                        menuIsCollapsed,
                        __typename: 'menu'
                    }
                };
&nbsp;
<span class="cstat-no" title="statement not covered" >                cache.writeData({ data });</span>
<span class="cstat-no" title="statement not covered" >                return null;</span>
            }
        }
    }
});
&nbsp;
const client = new ApolloClient({
    cache,
    link: ApolloLink.from([stateLink, new HttpLink({ uri: config.graphQlUrl })])
});
&nbsp;
client.onResetStore(stateLink.writeDefaults);
&nbsp;
export default client;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 13 2018 12:28:41 GMT+0000 (GMT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
