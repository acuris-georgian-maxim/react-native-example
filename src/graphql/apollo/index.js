// @flow

// $FlowFixMe
import { ApolloProvider, Query } from 'react-apollo';

// $FlowFixMe
import { compose, graphql } from 'react-apollo';

export { ApolloProvider };
export { Query };
export { compose };
export { graphql };

export { default as client } from './client';
