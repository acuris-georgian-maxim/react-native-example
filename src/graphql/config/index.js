// @flow

export { default as config } from './config';
