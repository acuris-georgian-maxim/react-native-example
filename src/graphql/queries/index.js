// @flow

export { default as serverQueries } from './server';
export { default as linkStateQueries } from './linkState';
