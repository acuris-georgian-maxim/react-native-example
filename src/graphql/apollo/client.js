// @flow

import { ApolloClient } from 'apollo-client';
import { withClientState } from 'apollo-link-state';
import { ApolloLink } from 'apollo-link';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { HttpLink } from 'apollo-link-http';

import { config } from '../config';

const cache = new InMemoryCache({
    dataIdFromObject: object => object.key || null
});

const defaults = {
    list: {
        listIsVisible: true,
        __typename: 'list'
    },
    menu: {
        menuIsCollapsed: false,
        __typename: 'menu'
    }
};

const stateLink = withClientState({
    cache,
    defaults,
    resolvers: {
        Mutation: {
            /* eslint-disable */
            updateListVisibility: (_, { listIsVisible }, { cache }) => {
                /* eslint-enabled */
                const data = {
                    list: {
                        listIsVisible,
                        __typename: 'list'
                    }
                };

                cache.writeData({ data });
                return null;
            },
            updateMenuVisiblity: (_, { menuIsCollapsed }, { cache }) => {
                /* eslint-enabled */
                const data = {
                    menu: {
                        menuIsCollapsed,
                        __typename: 'menu'
                    }
                };

                cache.writeData({ data });
                return null;
            }
        }
    }
});

const client = new ApolloClient({
    cache,
    link: ApolloLink.from([stateLink, new HttpLink({ uri: config.graphQlUrl })])
});

client.onResetStore(stateLink.writeDefaults);

export default client;
