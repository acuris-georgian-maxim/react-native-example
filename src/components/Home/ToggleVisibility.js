// @flow

import React, { Component } from 'react';
import { Button } from 'react-native';

import { graphql } from '../../graphql/apollo';
import { linkStateQueries } from '../../graphql/queries';

type State = {
    visible: boolean
};

type Props = {
    mutate: any
};

export class ToggleVisibility extends Component<Props, State> {
    state = {
        visible: true
    };

    onPress = (visible: boolean) => {
        this.props
            .mutate({ variables: { listIsVisible: visible } })
            .then(res => {
                if (res) {
                    this.setState({ visible });
                }
            });
    };

    render() {
        return (
            <Button
                color="#841584"
                onPress={() => this.onPress(!this.state.visible)}
                title={this.state.visible ? 'Hide list' : 'Show list'}
            />
        );
    }
}

const ToggleVisibilityWithMutation = graphql(
    linkStateQueries.updateListVisibility
)(ToggleVisibility);

export default ToggleVisibilityWithMutation;
