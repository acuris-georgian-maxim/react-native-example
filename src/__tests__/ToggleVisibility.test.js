import React from 'react';
import { shallow } from 'enzyme';

import { ApolloProvider } from 'react-apollo';
import { client } from '../graphql/apollo';

import ToggleVisibility from '../components/Home/ToggleVisibility';

describe('ToggleVisibility', () => {
  describe('render', () => {
    it('renders the component', () => {
      const tree = shallow(
        <ApolloProvider client={client}>
          <ToggleVisibility />
        </ApolloProvider>
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
