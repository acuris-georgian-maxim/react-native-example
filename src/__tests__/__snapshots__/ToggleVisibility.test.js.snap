// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleVisibility render renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
          },
          "data": DepTrackingCache {
            "data": Object {
              "$ROOT_QUERY.list": Object {
                "__typename": "list",
                "listIsVisible": true,
              },
              "$ROOT_QUERY.menu": Object {
                "__typename": "menu",
                "menuIsCollapsed": false,
              },
              "ROOT_QUERY": Object {
                "list": Object {
                  "generated": true,
                  "id": "$ROOT_QUERY.list",
                  "type": "id",
                  "typename": "list",
                },
                "menu": Object {
                  "generated": true,
                  "id": "$ROOT_QUERY.menu",
                  "type": "id",
                  "typename": "menu",
                },
              },
            },
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimistic": Array [],
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "keyMaker": QueryKeyMaker {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "perQueryKeyMakers": Map {},
            },
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {
            Object {
              "definitions": Array [
                Object {
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GeneratedClientQuery",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "list",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "listIsVisible",
                              },
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "__typename",
                              },
                            },
                          ],
                        },
                      },
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "menu",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "menuIsCollapsed",
                              },
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "__typename",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
              "kind": "Document",
            } => Object {
              "definitions": Array [
                Object {
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GeneratedClientQuery",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "list",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "listIsVisible",
                              },
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "__typename",
                              },
                            },
                          ],
                        },
                      },
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "menu",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "menuIsCollapsed",
                              },
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "__typename",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
              "kind": "Document",
            },
            Object {
              "definitions": Array [
                Object {
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GeneratedClientQuery",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "list",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "listIsVisible",
                              },
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "__typename",
                              },
                            },
                          ],
                        },
                      },
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "menu",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "menuIsCollapsed",
                              },
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "__typename",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
              "kind": "Document",
            } => Object {
              "definitions": Array [
                Object {
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "GeneratedClientQuery",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "list",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "listIsVisible",
                              },
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "__typename",
                              },
                            },
                          ],
                        },
                      },
                      Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "menu",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "menuIsCollapsed",
                              },
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "__typename",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
              "kind": "Document",
            },
          },
          "watches": Set {},
        },
        "clientAwareness": Object {},
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": ApolloLink {
          "request": [Function],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [
          [Function],
        ],
        "ssrMode": false,
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
            },
            "data": DepTrackingCache {
              "data": Object {
                "$ROOT_QUERY.list": Object {
                  "__typename": "list",
                  "listIsVisible": true,
                },
                "$ROOT_QUERY.menu": Object {
                  "__typename": "menu",
                  "menuIsCollapsed": false,
                },
                "ROOT_QUERY": Object {
                  "list": Object {
                    "generated": true,
                    "id": "$ROOT_QUERY.list",
                    "type": "id",
                    "typename": "list",
                  },
                  "menu": Object {
                    "generated": true,
                    "id": "$ROOT_QUERY.menu",
                    "type": "id",
                    "typename": "menu",
                  },
                },
              },
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimistic": Array [],
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "cacheKeyRoot": CacheKeyNode {
                "children": null,
                "key": null,
              },
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "keyMaker": QueryKeyMaker {
                "cacheKeyRoot": CacheKeyNode {
                  "children": null,
                  "key": null,
                },
                "perQueryKeyMakers": Map {},
              },
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {
              Object {
                "definitions": Array [
                  Object {
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "GeneratedClientQuery",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "list",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "listIsVisible",
                                },
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "__typename",
                                },
                              },
                            ],
                          },
                        },
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "menu",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "menuIsCollapsed",
                                },
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "__typename",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
                "kind": "Document",
              } => Object {
                "definitions": Array [
                  Object {
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "GeneratedClientQuery",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "list",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "listIsVisible",
                                },
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "__typename",
                                },
                              },
                            ],
                          },
                        },
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "menu",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "menuIsCollapsed",
                                },
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "__typename",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
                "kind": "Document",
              },
              Object {
                "definitions": Array [
                  Object {
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "GeneratedClientQuery",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "list",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "listIsVisible",
                                },
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "__typename",
                                },
                              },
                            ],
                          },
                        },
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "menu",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "menuIsCollapsed",
                                },
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "__typename",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
                "kind": "Document",
              } => Object {
                "definitions": Array [
                  Object {
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "GeneratedClientQuery",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "list",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "listIsVisible",
                                },
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "__typename",
                                },
                              },
                            ],
                          },
                        },
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "menu",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "menuIsCollapsed",
                                },
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "__typename",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
                "kind": "Document",
              },
            },
            "watches": Set {},
          },
        },
        "version": "2.4.7",
        "watchQuery": [Function],
      }
    }
  >
    <Apollo(ToggleVisibility) />
  </ApolloProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
